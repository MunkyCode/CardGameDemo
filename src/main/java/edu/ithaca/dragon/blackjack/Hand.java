package edu.ithaca.dragon.blackjack;

import java.util.ArrayList;
import java.util.Collections;

public class Hand {
    private ArrayList<Card> cards;

    public Hand(ArrayList<Card> startingCards) throws IllegalArgumentException{
        cards = new ArrayList<>();
        ArrayList<Card> toSort = (ArrayList<Card>)startingCards.clone();
        Collections.sort(toSort);
        for(int x = 0; x < toSort.size() -1; x ++){
            if(toSort.get(x).compareTo(toSort.get(x + 1)) == 0)
                throw new IllegalArgumentException("Cannot have duplicate cards in hand");
        }
        cards = (ArrayList<Card>)startingCards.clone();
    }
    public Hand(){
        cards = new ArrayList<>();
    }

    public ArrayList<Card> getCards() {
        return cards;
    }

    public int numCards(){return cards.size();}

    public void addCard(Card cardIn) throws IllegalArgumentException{
        if(cardIn == null) throw new IllegalArgumentException("Cannot add null card to hand");
        for(Card c: cards){
            if(c.compareTo(cardIn) == 0)
                throw new IllegalArgumentException("Cannot add duplicate card [" + c + "] to hand: " + this);
        }
        cards.add(cardIn);
    }

    public void clearCards(){
        cards.clear();
    }

    @Override
    public String toString() {
        return cards.toString();
    }
}
