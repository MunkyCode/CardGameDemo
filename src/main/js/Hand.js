import Card from "react-bootstrap/Card";
import CardGroup from "react-bootstrap/CardGroup";
import React from "react";
import {Container} from "react-bootstrap";

export default function Hand(props) {
    //using react-bootstrap data structure called Card, not to be confused with also talking about playing cards
    const cardsArray = props.cards.map(aCardModel => (
        <Card key={""+aCardModel.value+aCardModel.suit}>
            <Card.Title>{aCardModel.value}</Card.Title>
            <Card.Title> of</Card.Title>
            <Card.Title>{aCardModel.suit}</Card.Title>
        </Card>
    ));
    return (
        <Container>
            {props.ownerName}
            <CardGroup className="cardDeck">{cardsArray}</CardGroup>
        </Container>

    );
}