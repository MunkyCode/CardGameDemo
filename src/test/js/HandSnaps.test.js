import renderer from "react-test-renderer";
import React from "react";

import Hand from "../../main/js/Hand";
import {regHandEx, bigHandEx, singleCardHandEx} from "./Hand.examples";

test('regHandEx snapshots test', () => {
    const component = renderer.create(<Hand cards={regHandEx}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('bigHandEx snapshots test', () => {
    const component = renderer.create(<Hand cards={bigHandEx}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('singleCardHandEx snapshots test', () => {
    const component = renderer.create(<Hand cards={singleCardHandEx}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('noCardHandEx snapshots test', () => {
    const component = renderer.create(<Hand cards={[]}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});