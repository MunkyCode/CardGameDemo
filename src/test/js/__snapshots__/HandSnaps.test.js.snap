// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bigHandEx snapshots test 1`] = `
<div
  className="cardDeck card-group"
>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      2
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      spades
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      4
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      hearts
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      2
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      diamonds
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      5
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      hearts
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      3
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      hearts
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      2
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      clubs
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      2
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      hearts
    </div>
  </div>
</div>
`;

exports[`noCardHandEx snapshots test 1`] = `
<div
  className="cardDeck card-group"
/>
`;

exports[`regHandEx snapshots test 1`] = `
<div
  className="cardDeck card-group"
>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      6
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      diamonds
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      4
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      hearts
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      8
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      clubs
    </div>
  </div>
</div>
`;

exports[`singleCardHandEx snapshots test 1`] = `
<div
  className="cardDeck card-group"
>
  <div
    className="card"
  >
    <div
      className="card-title h5"
    >
      2
    </div>
    <div
      className="card-title h5"
    >
       of
    </div>
    <div
      className="card-title h5"
    >
      spades
    </div>
  </div>
</div>
`;
